adapters/repos/db/index_integration_test.go

```diff
@@ -0,0 +1,71 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2020 SeMI Technologies B.V. All rights reserved.
+//
+//  CONTACT: hello@semi.technology
+//
+
+// +build integrationTest
+
+package db
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/semi-technologies/weaviate/entities/schema"
+	"github.com/semi-technologies/weaviate/entities/schema/kind"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func getIndexFilenames(dirName string, className string) ([]string, error) {
+	filenames := []string{}
+	infos, err := ioutil.ReadDir(dirName)
+	if err != nil {
+		return filenames, err
+	}
+	for _, i := range infos {
+		if strings.Contains(i.Name(), className) {
+			filenames = append(filenames, i.Name())
+		}
+	}
+	return filenames, nil
+}
+
+func TestDropIndex(t *testing.T) {
+	rand.Seed(time.Now().UnixNano())
+	dirName := fmt.Sprintf("./testdata/%d", rand.Intn(10000000))
+	os.MkdirAll(dirName, 0o777)
+	defer func() {
+		err := os.RemoveAll(dirName)
+		fmt.Println(err)
+	}()
+	testClassName := "deletetest"
+	index, err := NewIndex(IndexConfig{
+		RootPath: dirName, Kind: kind.Thing, ClassName: schema.ClassName(testClassName),
+	}, &fakeSchemaGetter{}, nil, nil)
+	require.Nil(t, err)
+
+	indexFilesBeforeDelete, err := getIndexFilenames(dirName, testClassName)
+	require.Nil(t, err)
+
+	// drop the index
+	err = index.drop()
+	require.Nil(t, err)
+
+	indexFilesAfterDelete, err := getIndexFilenames(dirName, testClassName)
+	require.Nil(t, err)
+
+	assert.Equal(t, 3, len(indexFilesBeforeDelete))
+	assert.Equal(t, 0, len(indexFilesAfterDelete))
```

Comment:
I think we should extend this test (or add another one) to verify that

1. After an index has been deleted, it can be created with the same name again
2. The re-created index contains no objects even if it did prior to deletion
3. We can import data to the re-created index
4. We can perform inverted-index searches on the re-created index (any, simple search should prove that it works)
5. We can perform vector searches on the re-created index.

Reply:
I'll extend those tests